### PR TITLE
[rand_gen] add storage

### PR DESCRIPTION
**Pull request by zekun000**

### CODE REVIEW

**Comment by zekun000** [comment]
Current dependencies on/for this PR:
* main
  * **PR #11208** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11208?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #11217** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11217?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #11229** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11229?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11229?utm_source=stack-comment).

File: consensus/src/rand/rand_gen/storage/interface.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::rand::rand_gen::types::{
+    AugData, AugDataId, CertifiedAugData, RandDecision, RandShare, ShareId,
+};
+use aptos_consensus_types::randomness::RandMetadata;
+
+pub trait RandStorage<S, P, D> {
```
Comment (danielxiangzl) [inline consensus/src/rand/rand_gen/storage/interface.rs:9]: Should it be GCed every epoch?

**Review by grao1991 — APPROVED** [review]

File: consensus/src/rand/rand_gen/types.rs [hunk 2]
```diff
@@ -64,9 +68,19 @@ pub trait Proof: Clone + Send + Sync {
         Self: Sized;
 }
 
-pub trait AugmentedData: Clone + Send + Sync + Serialize {}
+pub trait AugmentedData:
```
[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/src/rand/rand_gen/types.rs:71]: I think it's required by the Schema

File: consensus/src/rand/rand_gen/storage/db.rs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    },
+};
+use anyhow::Result;
+use aptos_consensus_types::randomness::RandMetadata;
+use aptos_logger::info;
+use aptos_schemadb::{schema::Schema, Options, ReadOptions, SchemaBatch, DB};
+use std::{path::Path, sync::Arc, time::Instant};
+
+pub struct RandDb {
+    db: Arc<DB>,
```
[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/src/rand/rand_gen/storage/db.rs:27]: it'll be shared by multiple components to store data (I imagine)

**Pull request merged**